cmd/earthly/main.go

```diff
@@ -3205,6 +3329,257 @@ func (app *earthlyApp) actionListTargets(c *cli.Context) error {
+			return "", errors.Wrap(err, "failed finding org")
+		}
+		if len(orgs) != 1 {
+			return "", errors.New("more than one organizations available - please specify the name of the organization using `--org`")
+		}
+		app.satelliteOrg = orgs[0].Name
+		orgID = orgs[0].ID
+	} else {
+		var err error
+		orgID, err = cc.GetOrgID(app.satelliteOrg)
+		if err != nil {
+			return "", errors.Wrap(err, "invalid org provided")
+		}
+	}
+	return orgID, nil
+}
+
+func (app *earthlyApp) actionSatelliteLaunch(c *cli.Context) error {
+	app.commandName = "launch"
+
+	if c.NArg() != 1 {
+		return errors.New("satellite name is required")
+	}
+
+	app.satelliteName = c.Args().Get(0)
+
+	cc, err := cloud.NewClient(app.apiServer, app.sshAuthSock, app.authToken, app.console.Warnf)
+	if err != nil {
+		return errors.Wrap(err, "failed to create cloud client")
+	}
+
+	orgID, err := app.getSatelliteOrgID(cc)
+	if err != nil {
+		return err
+	}
+
+	app.console.Warnf("Launching Satellite. This could take a moment...")
+	_, err = cc.LaunchSatellite(app.satelliteName, orgID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create satellite %s", app.satelliteName)
+	}
+
+	err = app.useSatellite(c, app.satelliteName, orgID)
+	if err != nil {
+		return errors.Wrap(err, "could not configure satellite for use")
+	}
+
+	satellites, err := cc.ListSatellites(orgID)
+	if err != nil {
+		return err
+	}
+
+	app.printSatellites(satellites)
+	return nil
+}
+
+func (app *earthlyApp) actionSatelliteList(c *cli.Context) error {
+	app.commandName = "list"
+
+	if c.NArg() != 0 {
+		return errors.New("command does not accept any arguments")
+	}
+
+	cc, err := cloud.NewClient(app.apiServer, app.sshAuthSock, app.authToken, app.console.Warnf)
+	if err != nil {
+		return errors.Wrap(err, "failed to create cloud client")
+	}
+
+	orgID, err := app.getSatelliteOrgID(cc)
+	if err != nil {
+		return err
+	}
+
+	satellites, err := cc.ListSatellites(orgID)
+	if err != nil {
+		return err
+	}
+
+	app.printSatellites(satellites)
+	return nil
+}
+
+func (app *earthlyApp) actionSatelliteDestroy(c *cli.Context) error {
+	app.commandName = "launch"
+
+	if c.NArg() != 1 {
+		return errors.New("satellite name is required")
+	}
+
+	app.satelliteName = c.Args().Get(0)
+
+	cc, err := cloud.NewClient(app.apiServer, app.sshAuthSock, app.authToken, app.console.Warnf)
+	if err != nil {
+		return errors.Wrap(err, "failed to create cloud client")
+	}
+
+	orgID, err := app.getSatelliteOrgID(cc)
+	if err != nil {
+		return err
+	}
+
+	app.console.Warnf("Destroying Satellite. This could take a moment...")
+	err = cc.DeleteSatellite(app.satelliteName, orgID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete satellite %s", app.satelliteName)
+	}
+
+	if app.satelliteName == app.cfg.Satellite.Name {
+		// TODO what strategy do we want to use if you delete your current satellite?
+		if err = app.useSatellite(c, "", ""); err != nil {
+			return errors.Wrapf(err, "failed unselecting satellite")
+		}
```

Comment:
I would additionally `Printf` that the satellite was unselected as a result of the destruction, for clarity.